src/main/java/dev/openfeature/sdk/providers/memory/InMemoryProvider.java

```diff
@@ -47,88 +29,81 @@ public InMemoryProvider(Map<String, Flag<?>> flags) {
+     *
      * @param newFlag the flag to update
      */
     public void updateFlag(String flagKey, Flag<?> newFlag) {
         this.flags.put(flagKey, newFlag);
         ProviderEventDetails details = ProviderEventDetails.builder()
-            .flagsChanged(Collections.singletonList(flagKey))
-            .message("flag added/updated")
-            .build();
+                .flagsChanged(Collections.singletonList(flagKey))
+                .message("flag added/updated")
+                .build();
         emitProviderConfigurationChanged(details);
     }
 
     @Override
     public ProviderEvaluation<Boolean> getBooleanEvaluation(String key, Boolean defaultValue,
-            EvaluationContext evaluationContext) {
+                                                            EvaluationContext evaluationContext) {
         return getEvaluation(key, evaluationContext, Boolean.class);
     }
 
     @Override
     public ProviderEvaluation<String> getStringEvaluation(String key, String defaultValue,
-            EvaluationContext evaluationContext) {
+                                                          EvaluationContext evaluationContext) {
         return getEvaluation(key, evaluationContext, String.class);
     }
 
     @Override
     public ProviderEvaluation<Integer> getIntegerEvaluation(String key, Integer defaultValue,
-            EvaluationContext evaluationContext) {
+                                                            EvaluationContext evaluationContext) {
         return getEvaluation(key, evaluationContext, Integer.class);
     }
 
     @Override
     public ProviderEvaluation<Double> getDoubleEvaluation(String key, Double defaultValue,
-            EvaluationContext evaluationContext) {
+                                                          EvaluationContext evaluationContext) {
         return getEvaluation(key, evaluationContext, Double.class);
     }
 
     @SneakyThrows
     @Override
     public ProviderEvaluation<Value> getObjectEvaluation(String key, Value defaultValue,
-            EvaluationContext evaluationContext) {
+                                                         EvaluationContext evaluationContext) {
         return getEvaluation(key, evaluationContext, Value.class);
     }
 
     private <T> ProviderEvaluation<T> getEvaluation(
             String key, EvaluationContext evaluationContext, Class<?> expectedType
     ) throws OpenFeatureError {
-        if (!ProviderState.READY.equals(state)) {
-            if (ProviderState.NOT_READY.equals(state)) {
+        if (!ProviderState.READY.equals(getState())) {
+            if (ProviderState.NOT_READY.equals(getState())) {
                 throw new ProviderNotReadyError("provider not yet initialized");
             }
+            if (ProviderState.FATAL.equals(getState())) {
+                throw new FatalError("provider in fatal error state");
+            }
             throw new GeneralError("unknown error");
```

Comment:
The behaviour here is different than that of the `OpenFeatureClient`, which would not throw this `GeneralError`. Which bevaviour is the expected one?

Reply:
See: https://github.com/open-feature/java-sdk/pull/1096/files/98c620f14a81a0fda47207eb9d912826a33ec446#r1752386247

We should be able to delete this code entirely since initialization is tracked in the SDK, but things should also work fine if we don't touch this class at all.

Reply:
so do you suggest to generate a Second deprecated InMemoryProvider to always ensure backwards compatibility and run the tests for both? or do you think that is overkill?